File: app/src/main/java/com/starsep/sokoban/gamelogic/LevelConverterScript.java
package com.starsep.sokoban.gamelogic;

import java.util.ArrayList;
import java.util.Scanner;

public class LevelConverterScript {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		ArrayList<String> data = new ArrayList<>();
		while (input.hasNextLine()) {
			String line = input.nextLine();
			if (line != null) {
				data.add(line);
			}
		}
		int height = data.size();
		int width = data.get(0).length();
		char[] level = new char[height * width];
		for (int i = 0; i < height; i++) {
			for (int j = 0; j < width; j++) {
				level[i * width + j] = data.get(i).charAt(j);
			}
		}
		try {
			Level l = LevelConverter.convert(level, width);
			System.out.print(l);
		} catch (LevelConverter.LevelConverterException e) {
			e.printStackTrace();
		}
		input.close();
	}
}

File: app/src/main/kotlin/com.starsep.sokoban.release/gamelogic/level/LevelConverterScript.kt
package com.starsep.sokoban.release.gamelogic.level

import java.util.ArrayList
import java.util.Scanner

object LevelConverterScript {
    @JvmStatic
    fun main(args: Array<String>) {
        val input = Scanner(System.`in`)
        val data = ArrayList<String>()
        while (input.hasNextLine()) {
            val line = input.nextLine()
            if (line != null) {
                data.add(line)
            }
        }
        val height = data.size
        val width = data[0].length
        val level = CharArray(height * width)
        for (i in 0 until height) {
            for (j in 0 until width) {
                level[i * width + j] = data[i][j]
            }
        }
        try {
            val l = LevelConverter.convert(level, width)
            print(l)
        } catch (e: LevelConverter.LevelConverterException) {
            e.printStackTrace()
        }

        input.close()
    }
}
